### PR TITLE
Tech/add annotation clustering

### PR DESCRIPTION
**Pull request by amarchenko25**

### CODE REVIEW

File: lib/src/annotation/point_annotation_manager.dart [hunk 1]
```diff
@@ -7,6 +7,9 @@ class PointAnnotationManager extends BaseAnnotationManager {
       {required String id, required BinaryMessenger messenger})
       : super(id: id, messenger: messenger);
 
+  /// Needs for handling tap on cluster
+  String get clusterLayerId => 'mapbox-iOS-cluster-circle-layer-manager-$id';
```
Comment (river2202) [inline lib/src/annotation/point_annotation_manager.dart:11]: This feels like should be done in the application layer (App / example app) keep flutter sdk platform agnostic.

Refer to iOS and flutter sdk examples, these kinds of hardcoded value are more like un-official constants that available to developer so SDK shouldn't provide them. 

But since we are trying I think we can move on with it. And maybe move to helper extension in another file with both iOS and android constants in future.  For our own customisation, we should definitely make our own life as SDK consumer easier by providing these kinds of helper value. 

In summary, no action needed for now.

**Review by river2202 — APPROVED** [review]

**Comment by river2202** [comment]
I will leave it to you to merge the PR when you are ready. Feel free to push more reasonable changes.
